### ISSUE TITLE
Slow array function used in loop: Looping over something else then the array?

### ISSUE THREAD

**Issue by davidmosterd** [kalessil/phpinspectionsea#1713]
Description:
Say you are looping over something else than an array, WooCommerce orders in my case. I create an array with defaults, but on a certain condition, I want to add to these defaults another array. The inspection says it's slow to use an array in a loop, and if I were looping over this array I would understand this. But for associative arrays with strings there is no spread operator and typing out all these options is gross syntax.

It should maybe check array_merge is used in a loop that has nothing to do with the array itself? I see no other way of doing it with elegant syntax. Or am I missing something?

Example:
```
<?php

foreach( $orders as $order ) {
	$meta = [
		'thing_1' => 2,
		'thing_2' => 4
	];

	if ( $someCondition ) {
		$meta = array_merge( $meta, [
			'thing_1' => 3,
			'thing_5' => 4,
			'thing_8' => 9,
			'thing_13' => 5,
		] );
	}
}
```

**Comment by ea-inspections-team** [comment]
Thanks for reporting @davidmosterd ! Actually, that a valid concern - the inspection should not report this as the variable was assigned in the first statement of the loop. We'll fix this.

**Comment by davidmosterd** [comment]
Thanks! I disabled the inspection, but I'd rather keep it on ofc.

**Comment by reedy** [comment]
Come across this myself too - https://github.com/wikimedia/mediawiki-extensions-TorBlock/blob/e90ccfb/includes/TorExitNodes.php#L200-L204

```
		$nodes = [];
		foreach ( $data['relays'] as $relay ) {
			$addresses = $relay['or_addresses'];
			if ( isset( $relay['exit_addresses'] ) ) {
				$addresses = array_merge( $addresses, $relay['exit_addresses'] );
			}

			foreach ( $addresses as $ip ) {
```

**Comment by ea-inspections-team** [comment]
Thank you for reporting @davidmosterd and @reedy! The false-positive has been addressed.
